sklearn/tree/tests/test_tree.py

```diff
@@ -1984,7 +2025,7 @@ def assert_is_subtree(tree, subtree):
 
 def test_prune_tree_raises_negative_ccp_alpha():
     clf = DecisionTreeClassifier()
-    msg = "ccp_alpha must be greater than or equal to 0"
+    msg = "must be >= 0.0."
```

Comment:
since `_prune_tree` is called in `fit` I would expect `check_scalar` to be called before raising this error.